packages/salesforcedx-vscode-core/src/util/authInfo.ts

```diff
@@ -4,60 +4,53 @@
-  Aliases,
-  AuthInfo,
-  ConfigAggregator,
-  ConfigFile
-} from '@salesforce/core';
-import * as path from 'path';
+import { Aliases, AuthInfo } from '@salesforce/core';
+import { isUndefined } from 'util';
+import { channelService } from '../channels';
 import { nls } from '../messages';
-import { getRootWorkspacePath } from './index';
+import { notificationService } from '../notifications';
+import { ConfigSource, ConfigUtil } from './index';
+
+const defaultUserNameKey = 'defaultusername';
+const defaultDevHubUserNameKey = 'defaultdevhubusername';
 export class OrgAuthInfo {
   public static async getDefaultUsernameOrAlias(): Promise<string | undefined> {
-    try {
-      const rootPath = getRootWorkspacePath();
-      const myLocalConfig = await ConfigFile.create({
-        isGlobal: false,
-        rootFolder: path.join(rootPath, '.sfdx'),
-        filename: 'sfdx-config.json'
-      });
-      const localDefault = myLocalConfig.get('defaultusername');
-      return JSON.stringify(localDefault).replace(/\"/g, '');
-    } catch {
-      await this.getGlobalDefaults('defaultusername');
-    }
-  }
-
-  private static async getGlobalDefaults(usernameType: string) {
-    try {
-      const aggregator = await ConfigAggregator.create();
-      const globalDefault = aggregator.getPropertyValue(usernameType);
-      return JSON.stringify(globalDefault).replace(/\"/g, '');
-    } catch {
-      if (usernameType === 'defaultusername') {
-        console.error(nls.localize('error_no_default_username'));
-      } else {
-        console.error(nls.localize('error_ no_default_devhubusername'));
+    const defaultUserName = await ConfigUtil.getConfigValue(defaultUserNameKey);
+    if (isUndefined(defaultUserName)) {
+      displayMessage(
+        nls.localize('error_no_default_username'),
+        true,
+        VSCodeWindowTypeEnum.Informational
+      );
+      return undefined;
+    } else {
+      const configSource = await ConfigUtil.getConfigSource(defaultUserNameKey);
+      if (configSource === ConfigSource.Global) {
+        displayMessage(
+          nls.localize('warning_using_global_username'),
+          true,
+          VSCodeWindowTypeEnum.Warning
+        );
       }
     }
+    return JSON.stringify(defaultUserName).replace(/\"/g, '');
   }
 
   public static async getDefaultDevHubUsernameOrAlias(): Promise<
     string | undefined
   > {
-    try {
-      const rootPath = getRootWorkspacePath();
-      const myLocalConfig = await ConfigFile.create({
-        isGlobal: false,
-        rootFolder: path.join(rootPath, '.sfdx'),
-        filename: 'sfdx-config.json'
-      });
-      const localDefault = myLocalConfig.get('defaultdevhubusername');
-      return JSON.stringify(localDefault).replace(/\"/g, '');
-    } catch {
-      await this.getGlobalDefaults('defaultdevhubusername');
+    const defaultDevHubUserName = await ConfigUtil.getConfigValue(
+      defaultDevHubUserNameKey
+    );
+    if (isUndefined(defaultDevHubUserName)) {
```

Comment:
I like that we're displaying the message for `defaultUsername` and `defaultDevHubUsername`, but I was wondering why the message is informational for the `username` and an error for the `devhubusername`?

Reply:
Prior to me touching anything in this class it was always writing to console.error for both username and devhubusername. For the defaultusername, error wasn't correct because that's something that a user would see when they create a project or load a project that's been created but not authenticated or used to create an org so informational was correct there. For the case where we're pulling defaultusername from the global config, that's definitely a warning. The debhubusername isn't part of the bug or something I vetted, that was always being output as an error so I left that unchanged.

Reply:
Ah okay makes sense, so the missing devhubusername error was originally being reported using console.error as well, so the user never actually saw it as a pop up. Since surfacing this error to the user isn't part of this bug, are we would still like to display this error using the vscode pop up too?

Reply:
It seems that if there's an actual error that they should have been notifications to begin with instead of just writing to the console since that's the pattern we've adopted. The way I wrote the function, if people flip the other direction and don't want notifications, the notification can be turned off with a boolean

Reply:
I agree, I think it makes sense to display the notification to the user. But since we are displaying a notification to the user in case of a missing default username and missing default dev hub username, should we try to make them the same type of notification? As in, should both of them be informational? Or both errors? I'm just trying to understand whether we want the same sort of behavior in both scenarios.

Reply:
The informational message (no default user in local or global) will get shown when the user first creates a project or if the user loads a project that hasn't been authenticated yet. The fact that it was originally an error message was just simply incorrect. The warning message is because we found a default user but it was found in the global which means you could possibly be changing something in an org that you didn't want to change it. This message should have always been a warning and not an error. Finally, the devhub username doesn't actually enter into this equation for my changes. it was previously an error in the code and I didn't change that. I didn't fully vet the scenario in which this message is displayed but kept it as it was. If the message type when the devhub username isn't found is not correct then that is a change separate from this.